### DISCUSSION TITLE
Standalone application

### DISCUSSION THREAD

**Discussion by Cortexico** [adamerose/pandasgui#248] [General]
Hello,

I'm working on PandasGui and found out there is a lot of extern dependencies and calls. Also I couldn't find a way to make Plotly work correctly offline, so I'm now working on a full integration of interactive Matplotlib plots instead. I'm encountering a lot of challenges as there are some limitations with it and the interactive character I want to maintain. 
So I would like to know if anyone already tried something similar ? Or have any Idea on how to build correctly the figure_viewer for the Matplotlib integration ?

If any of you have a better idea than Matplotlib, I was also thinking of Seaborn but I also need to consider dependencies as this may be used as a package later on.

**Comment by adamerose** [comment]
> Also I couldn't find a way to make Plotly work correctly offline

The way PandasGUI uses Plotly doesn't involve the internet, it just saves an HTML file to your local temp directory and displays it in a webview through PyQt. If you care about interactivity I think that rules out Matplotlib and Seaborn, unless you only want to zoom and pan without highlighting points or anything related to the data.

> If any of you have a better idea than Matplotlib, I was also thinking of Seaborn but I also need to consider dependencies as this may be used as a package later on.

Seaborn is just a wrapper around Matplotlib, it's the same underlying engine. The strongest alternatives to Plotly I was considering before settling on it for PandasGUI were Bokeh and Altair, which also use HTML and JS under the hood for rendering.
